### PR TITLE
build: add Semnatic-PR.yml gh action

### PR DESCRIPTION
**Pull request by sr2706**
Added Semnatic PR github action

### CODE REVIEW

File: .github/workflows/Semnatic-PR.yml [hunk 1]
```diff
@@ -0,0 +1,17 @@
+name: Lint PR
+'on':
```
Comment (ad1992) [inline .github/workflows/Semnatic-PR.yml:2]:
```suggestion
on:
```

File: .github/workflows/Semnatic-PR.yml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+name: Lint PR
```
Comment (ad1992) [inline .github/workflows/Semnatic-PR.yml:1]:
```suggestion
name: Semantic PR title
```

File: .github/workflows/Semnatic-PR.yml [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+  pull_request_target:
+    types:
+      - opened
+      - edited
+      - synchronize
+permissions:
+  pull-requests: read
+jobs:
+  main:
+    name: Validate PR title
```
Comment (ad1992) [inline .github/workflows/Semnatic-PR.yml:12]:
```suggestion
```
not needed

**Review by ad1992 — CHANGES_REQUESTED** [review]

File: .github/workflows/Semnatic-PR.yml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+    runs-on: ubuntu-latest
+    steps:
+      - uses: amannn/action-semantic-pull-request@v5
+        env:
+          GITHUB_TOKEN: '${{ secrets.GITHUB_TOKEN }}'
```
Comment (ad1992) [inline .github/workflows/Semnatic-PR.yml:17]:
```suggestion
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```
quotes shouldn't be needed

**Review by ad1992 — CHANGES_REQUESTED** [review]

**Comment by sr2706** [comment]
Hey @ad1992 I have changed all your suggestions can you merge this

**Review by ad1992 — APPROVED** [review]
Thanks @sr2706 ✨

**Pull request merged**